fapi-client/src/main/scala/com/gu/facia/api/models/curatedcontent.scala

```diff
@@ -4,56 +4,47 @@ import com.gu.contentapi.client.model.Content
 import com.gu.facia.api.utils.{ResolvedMetaData, ItemKicker}
 import com.gu.facia.client.models.{SupportingItem, MetaDataCommonFields, Trail, TrailMetaData}
 
-case class ImageReplace(imageSrc: String, imageSrcWidth: String, imageSrcHeight: String)
-
-object ImageReplace {
-  def fromTrailMeta(trailMeta: MetaDataCommonFields): Option[ImageReplace] =
-    for {
-      imageReplace <- trailMeta.imageReplace.filter(identity)
-      imageSrc <- trailMeta.imageSrc
-      imageSrcWidth <- trailMeta.imageSrcWidth
-      imageSrcHeight <- trailMeta.imageSrcHeight
-    } yield ImageReplace(imageSrc, imageSrcWidth, imageSrcHeight)
-}
+case class FaciaImage(
+  imageType: String,
+  imageSrc: String,
+  imageSrcWidth: Option[String],
+  imageSrcHeight: Option[String]
+)
+
+object FaciaImage {
+
+  def getFaciaImage(maybeContent: Option[Content], trailMeta: MetaDataCommonFields): Option[FaciaImage] = {
+    if (trailMeta.imageHide.exists(identity)) None
+    else { maybeContent flatMap { content =>
+        val resolvedMetadata = ResolvedMetaData.fromContentAndTrailMetaData(content, trailMeta)
```

Comment:
Can you merge in `master`, resolve the conflicts and pass this in? One of the things I have done is do this resolving once, then pass along the resolved instance to where you need it.